letta/client/client.py

```diff
@@ -428,9 +429,18 @@ def create_agent(
             agent_type=agent_type,
             llm_config=llm_config if llm_config else self._default_llm_config,
             embedding_config=embedding_config if embedding_config else self._default_embedding_config,
+            initial_message_sequence=initial_message_sequence,
+        )
+
+        # Use model_dump_json() instead of model_dump()
```

Comment:
@sarahwooders I was getting a JSON serialization error here caused by the `datetime` inside of `Message.created_by` not getting serialized properly

It looks like the fix is to use `model_dump_json` instead of `model_dump`, to force the conversion to happen via `pydantic` instead of `requests` (`pydantic` understands that it needs to convert to ISO format because of the `to_json` inside of `Message`, but `requests` don't know how to use that)